src/main/scala/io/kx/loanapp/action/LoanAppEventingToProcAction.scala

```diff
@@ -28,13 +28,13 @@ class LoanAppEventingToProcAction(creationContext: ActionCreationContext) extend
         submitted.loanDurationMonths
       )).execute()
         .map { _ =>
-          effects.reply(Empty.defaultInstance)
+          Empty.defaultInstance
         }
         .recover{ ex =>
           log.error("onSubmitted error [{}]: {}",submitted.loanAppId,ex)
-          effects.reply(Empty.defaultInstance)
+          Empty.defaultInstance
         }
-    effects.asyncEffect(futureEffect)
+    effects.asyncReply(futureEffect)
```

Comment:
we don't need to wrap everything in an Effect. We can map to the response type `Empty` and then pass the `Future[Empty]` to `effects.asyncResult`